index.js

```diff
@@ -1270,78 +947,15 @@
                    rejectInconsistent);
     };
 
-    //  paramTypesAndValues :: ((a -> Any), Type, [a]) -> ErrorRepr
-    var paramTypesAndValues = function(valFn, type, values) {
-      return map(
-        chain(values, valFn),
-        function(value) {
-          return typesAndValues(
-            type,
-            [value]
-          );
-        }
-      );
-    };
-
-    //  nullaryTypesAndValues :: [Any] -> ErrorRepr
-    var nullaryTypesAndValues = function(values) {
-      return {
-        isParameterizedType: false,
-        types: valueTypes(values),
-        values: values
-      };
-    };
-
-    //  typesAndValues :: (type, [Any]) -> ErrorRepr
-    var typesAndValues = function(t, values) {
-      if (isParameterizedType(t)) {
-        var constructorName = functionName(values[0].constructor);
```

Comment:
These are the three issues with which we're faced:

```javascript
Nullable a          'XXX'               String( "XXX" :: String )
         ^

Identity a          Identity('XXX')     Object( "XXX" :: String )
         ^

Either a b          Left('XXX')         Either( "XXX" :: String )
       ^
```

This first could be fixed with a special case for nullable types. I've no problem with this as nullable types are weird and already require special treatment to avoid infinite recursion when determining a value's types.

The second issue could be fixed, as you suggest, by using the unprefixed type identifier (which could come from the type, I believe, so we needn't rely on the `@@type` property). For types with multiple data constructors, this would bring the third issue into play.

The third issue could be fixed, as you suggest, by requiring that all values provide a suitable `constructor.name`. This doesn't sit well with me. It suggests that values should be created via object-oriented constructor functions. This may lead people to view Left and Right, say, as subtypes of Either, an abstract base type. Inheritance muddies the conversation.

The other problem with constructor functions is they facilitate the use of `instanceof`, which I consider harmful. One could, of course, include `constructor: {name: 'Identity'}` in a value to "trick" sanctuary-def, but this strikes me as quite confusing.

I'd prefer to consult another `@@something` property than to use `constructor.name`.

It appears not to be possible to display context *consistently* without introducing another convention (such as `constructor.name` or `@@something`). Given that we already underline the relevant type variables, I don't think a new convention is warranted. What do you think, @kedashoe?